Sources/ZcashLightClientKit/Utils/SaplingParameterDownloader.swift

```diff
@@ -61,52 +74,37 @@ public enum SaplingParameterDownloader {
 
         task.resume()
     }
-    /**
-    Downloads the parameters if not present and provides the resulting URLs for both parameters
-    - Parameters:
-        - spendURL: URL to check whether the parameter is already downloaded
-        - outputURL: URL to check whether the parameter is already downloaded
-        - result: block to handle success or error
-    */
+    
+    /// Downloads the parameters if not present and provides the resulting URLs for both parameters
+    /// - Parameters:
+    ///     - spendURL: URL to check whether the parameter is already downloaded
+    ///     - outputURL: URL to check whether the parameter is already downloaded
     public static func downloadParamsIfnotPresent(
         spendURL: URL,
-        outputURL: URL,
-        result: @escaping (Result<(spend: URL, output: URL), Error>) -> Void
-    ) {
-        ensureSpendParameter(at: spendURL) { spendResult in
-            switch spendResult {
-            case .success(let spendResultURL):
-                ensureOutputParameter(at: outputURL) { outputResult in
-                    switch outputResult {
-                    case .success(let outputResultURL):
-                        result(.success((spendResultURL, outputResultURL)))
-                    case .failure(let outputResultError):
-                        result(.failure(Errors.failed(error: outputResultError)))
-                    }
-                }
-            case .failure(let spendResultError):
-                result(.failure(Errors.failed(error: spendResultError)))
-            }
+        outputURL: URL
+    ) async throws -> (spend: URL, output: URL) {
+        do {
+            let spendResultURL = try await ensureSpendParameter(at: spendURL)
+            let outputResultURL = try await ensureOutputParameter(at: outputURL)
+            return (spendResultURL, outputResultURL)
```

Comment:
This could be done in parallel
```suggestion
            async let spendResultURL = try ensureSpendParameter(at: spendURL)
            async let outputResultURL = try ensureOutputParameter(at: outputURL)
            let results =  await [spendResultURL, outputResultURL]
            return (result.0, result.1)
```

Reply:
https://www.avanderlee.com/swift/async-let-asynchronous-functions-in-parallel/

Reply:
agreed, implemented